test/probes/generic-pool.test.js

```diff
@@ -92,17 +102,17 @@ describe('probes/generic-pool ' + pkg.version, function () {
         }
         gpDebug('%s acquired(queue) %o for %e', span, resource, ao.lastEvent)
 
-        should.exist(ao.lastEvent.Layer)
-        ao.lastEvent.Layer.should.equal(span)
-        taskId.should.be.equal(ao.lastEvent.taskId)
+        expect(ao.lastEvent).exist;
+        expect(ao.lastEvent).property('Layer', span);
+        expect(ao.lastEvent).property('taskId', taskId);
 
         const t = setInterval(function () {
           if (okToRelease) {
-            gpDebug('releasing %o by %e', resource, ao.lastEvent)
+            gpDebug('%s releasing %o by %e', span, resource, ao.lastEvent)
 
-            should.exist(ao.lastEvent.Layer)
-            ao.lastEvent.Layer.should.equal(span)
-            taskId.should.be.equal(ao.lastEvent.taskId)
+            expect(ao.lastEvent, 'context when releasing').exist;
```

Comment:
Curious, are the `context when ...` internal properties of an event (and not sent when an event is actually reported to the collector)?

Reply:
no, 'context when releasing' is just the message that will be issued. without a message it would just report 'undefined' expected to exist which isn't very helpful.